subprojects/core-api/src/main/java/org/gradle/StartParameter.java

```diff
@@ -717,6 +719,26 @@ public void setBuildCacheDebugLogging(boolean buildCacheDebugLogging) {
         this.buildCacheDebugLogging = buildCacheDebugLogging;
     }
 
+    /**
```

Comment:
I'm not sure we want to expose this on `StartParameter` yet, at least not as a mutable property. By the time user code can see the `StartParameter` instance, any changes to the property will be ignored. For now, I'd maybe add this to `StartParameterInternal`

Another option would be to use the existing `InstantExecutionStartParameter` service that exposes some internal flags, such as the instant execution flags and options. For this, I would add a session (or build tree) scoped service that can take a `BuildOption` definition and calculate what value the build option has. This way, we're taking some small steps towards user defined build options.